demos/nddata_NNLS_Tikhonov_190104.py

```diff
@@ -18,268 +18,268 @@
+## 
+## unlike for the numpy version, I skip straight to the
+## vectorized/parallel version.
+#
+#
+#l = sqrt(logspace(-8,4,10)) # I do this because it gives me a fairly even spacing of points
+#test_data_2d = test_data * nddata(r_[1,1,1],r'\Omega')
+##@timeit
+#def multifreq_lcurve(l):
+#    return test_data_2d.C.nnls('t',
+#            R,lambda x,y: exp(-y*x), l=l)
+#x = multifreq_lcurve(l)
+#
+## 
+#
+#fl.next('L-curve')
+#chosen_l = 0.025
+#L_curve(l, x.get_prop('nnls_residual')[r'\Omega',0].data,
+#        x.C.run(linalg.norm,'R')[r'\Omega',0].data, markersize=5, alpha=0.5,
+#        show_l=chosen_l,
+#        label='compiled loop')
+#
+## and show the final result
+## here, I omit the SVD (allows negative) result
+#
+#
+#fl.next(r'show result where $\lambda$ set to knee')
+#test_data.nnls('t', R, lambda x,y: exp(-x*y), l=0.025)
+#fl.plot(test_data)
+#fl.plot(P)
+#
+#
+## ## 1.5D code -- a more complicated/realistic example
+## 
+## First, generate a test distribution, where we assume we have two dimensions:
+## 
+## *  A $t_2$ dimension, which is actually typically a frequency dimension
+## *  A $t_{indirect}$ dimension -- representing, *e.g.*, the delay of an inversion recovery experiment. 
+#
+#
+#time_endpoint = 0.2
+#R = nddata(r_[1.:100:100j],'R') # distribution of T2 relaxation rates
+#t2 = nddata(r_[1e-3:time_endpoint:2048j],'t2') # column vectors give functions of time
+#peaks_2D = [(20,1,0.8,0.025,5e-3,1),
+#           (30,1,0.8,0.075,1e-2,1),
+#           (80,5,1,0.175,1e-2,1)]
+#data_dist = 0
+#for x_mu,x_sigma,x_amp,y_mu,y_sigma,y_amp in peaks_2D:
+#        data_dist += x_amp*y_amp*exp(-(R-x_mu)**2/2/x_sigma**2
+#                           -(t2-y_mu)**2/2/y_sigma**2)
+#fl.next('Test 1.5D true distribution')
+#fl.image(data_dist)
+#
+#
+## Convert to the time domain, where the $R$ dimension is replaced by a
+## $t_{indirect}$ dimension, and add noise
+#
+#
+#n_indirect = 128
+#t_indirect = nddata((r_[0:n_indirect]+1)/double(n_indirect),
+#        't_indirect')
+#kernel = exp(-t_indirect*R)
+#test_data_2d = kernel.C.dot(data_dist)
+#fl.next('Test 1.5D dataset')
+#test_data_2d.add_noise(0.1)
+#fl.image(test_data_2d)
+#
+#
+## Generate the L-curve
+#
+#l = sqrt(logspace(-8,4,10))
+##@timeit
+#def multifreq_lcurve(l):
+#    return test_data_2d.C.nnls('t_indirect',
+#            R,lambda x,y: exp(-y*x), l=l)
+#x = multifreq_lcurve(l)
+#
+#
+## 
+#
+#
+#print(ndshape(test_data_2d))
+#print(ndshape(x))
+#
+#
+## 
+#
+#
+#fl.next('L-curve')
+#heel_lambda = 0.215
+#L_curve(l, x.get_prop('nnls_residual').C.sum('t2').data,
+#        x.C.run(linalg.norm,'R').sum('t2').data, markersize=5, alpha=0.5,
+#        label='compiled loop',
+#        show_l=heel_lambda, s=1e-2)
+#
+#
+## Final result for 1.5D test data
+#
+#fl.next(r'show result where $\lambda$ set to knee')
+#result = test_data_2d.C.nnls('t_indirect',R,
+#                    lambda x,y: exp(-y*x), l=heel_lambda)
+#fl.image(result)
+#
+#
+## ## Testing 2D BRD extension
```

Comment:
please don't check in commented code

I do think it's useful to have a test, but can we make this a script (in case this is a notebook) and put in the "examples" folder -- I would like to get rid of the demo folder

Reply:
I do not know exactly what this is or why it was commented out. But this code as it stands runs on my computer -- it takes some time because it is doing the timing thing that you had put in at some point a while back.

Reply:
I know @guinneak198 got stuff to work wrt demos/sphinx(?) so maybe if she and I have a brief meeting I can try to implement all that on what is here and the other items in the demos folder... there is also an examples folder, which makes things a little confusing

Reply:
that's a good idea, if it's a quick thing.  There should be no separate demos folder, I think, just an examples folder.  None of the demos should be intended for ipynb, only stand-alone scripts

Reply:
At this point I think we are more concerned with getting this PR merged for processing the ILT's of D2O data. I think examples could be a separate pull request/wrike task to organize things a bit better?

Reply:
This example takes forever to run. I'm adding it to examples and removing it from here but we should consider the time consumption